### PR TITLE
TimelineEvent : fix memory leak

### PR DESCRIPTION
**Pull request by ganfra**
I spotted this potential memory leak while digging performance issues.

### CODE REVIEW

**Comment by jmartinesp** [comment]
Those are actually all data classes, can they link something? They don't have an underlying pointer AFAIK.

They're mapped in:

```kotlin
public object FfiConverterTypeTimelineItemContentKind : FfiConverterRustBuffer<TimelineItemContentKind>{
    override fun read(buf: ByteBuffer): TimelineItemContentKind {
        return when(buf.getInt()) {
            1 -> TimelineItemContentKind.Message
            2 -> TimelineItemContentKind.RedactedMessage
            3 -> TimelineItemContentKind.Sticker(
                FfiConverterString.read(buf),
                FfiConverterTypeImageInfo.read(buf),
                FfiConverterString.read(buf),
            )
            ...
```

**Comment by ganfra** [comment]
Actually I'm not sure, as they implement Disposable interface, but yeah you're right, I don't know...

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/TimelineEventContentMapper.kt [hunk 1]
```diff
@@ -34,84 +34,91 @@ import io.element.android.libraries.matrix.impl.media.map
...
 
-    fun map(content: TimelineItemContent): EventContent = content.use {
-        when (val kind = it.kind()) {
-            is TimelineItemContentKind.FailedToParseMessageLike -> {
-                FailedToParseMessageLikeContent(
-                    eventType = kind.eventType,
-                    error = kind.error
-                )
+    fun map(content: TimelineItemContent): EventContent {
+        return content.use { _ ->
```
Comment (jmartinesp) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/TimelineEventContentMapper.kt:45]: It's a bit weird to ignore the parameter with `_` and then user the original value. I mean, it's valid, but a bit confusing.

**Review by jmartinesp — APPROVED** [review]
LGTM, thanks.

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/TimelineEventContentMapper.kt (hunk 1 above)
Comment (ganfra) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/TimelineEventContentMapper.kt:45]: Yeah but I don't like to use `it` and I can't find suitable name not confusing neither...

Comment (bmarty) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/TimelineEventContentMapper.kt:45]:
```suggestion
        return content.use {
```

works as well.

Comment (ganfra) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/TimelineEventContentMapper.kt:45]: You're right, less confusing.

**Pull request merged**